src/components/appMenuSections/dashboard/Dashboard.tsx

```diff
@@ -48,43 +97,70 @@ const Dashboard = () => {
     }
   }, [currentRole]);
 
+  useEffect(() => {
+    const getBlocksWon = async () => {
+      const blocks = await readOnlyGetBlocksWon();
+      setBlocksWon(blocks);
+    };
+    getBlocksWon();
+  }, [blocksWon]);
+
+  useEffect(() => {
+    const getStacksRewards = async () => {
+      const stacks = await readOnlyGetStacksRewards();
+      setStacksRewards(stacks);
+    };
+    getStacksRewards();
+  }, [stacksRewards]);
+
   return (
-    <div>
-      <h2>Dashboard</h2>
-      <h4>General info about Stacks - widgets/statistics</h4>
-      <ul>
-        <li>stacks rewards</li>
-        <li>miners</li>
-        <li>blocks</li>
-      </ul>
-      <h4>General info about mining pool</h4>
-      <ul>
-        <li>notifier</li>
-        <li>list of miners</li>
-        <li>winner block id</li>
-        <li>number of blocks won</li>
-        <li>stacks rewards</li>
-      </ul>
-      {currentRole !== 'Miner' && !clickedJoinPoolButtonByViewer && (
-        <Button
-          variant="contained"
-          onClick={() => {
-            if (currentRole === 'Viewer') {
-              handleJoinPoolButtonByViewer();
-            } else if (currentRole === 'NormalUser') {
-              handleJoinPoolButtonByNormalUser();
-            }
-          }}
-        >
-          Join pool
-        </Button>
-      )}
-      {clickedJoinPoolButtonByViewer && authenticatedSuccessfully === false && (
-        <LoadingButton loading loadingPosition="start" startIcon={<SaveIcon />} variant="outlined">
-          JOIN POOL
-        </LoadingButton>
-      )}
-    </div>
+    <Box
+      sx={{
+        minHeight: 'calc(100vh - 60px)',
+        backgroundColor: colors[currentTheme].accent2,
+        color: colors[currentTheme].secondary,
+        marginTop: -2.5,
+      }}
+    >
+      <div>
+        <h2>Dashboard</h2>
+        <h4>General info about mining pool</h4>
+        <ul>
+          <li>
+            notifier: <div>{currentNotifier}</div>
+          </li>
+          <li>
+            list of miners:
+            {minersList.map((data: string, index: number) => (
+              <div key={index}>{data}</div>
+            ))}
+          </li>
+          {currentRole === 'NormalUser' && <li>winner block id</li>}
+          {blocksWon !== null && <li>number of blocks won:{blocksWon} </li>}
+          {stacksRewards !== null && <li>stacks rewards:{stacksRewards} </li>}
+        </ul>
+        {currentRole !== 'Miner' && !clickedJoinPoolButtonByViewer && (
+          <Button
+            variant="contained"
+            onClick={() => {
+              if (currentRole === 'Viewer') {
+                handleJoinPoolButtonByViewer();
+              } else if (currentRole === 'NormalUser' || 'Miner') {
+                ContractAskToJoin(`${userAddress}`);
+                // handleJoinPoolButtonByNormalUser();
```

Comment:
Remove this function call and the implementation of it.

Reply:
it is resolved in a later commit from another branch. has the pr open.